src/main/java/sirius/biz/codelists/LookupTable.java

```diff
@@ -120,6 +137,85 @@ protected String normalizeCodeValue(String code) {
         return performFetchField(normalizeCodeValue(code), targetField);
     }
 
+    /**
+     * Fetches the mapping for the given code.
+     * <p>
+     * Mappings are most probably only supported by IDB backed tables. If <tt>acme</tt> is given, this will
+     * return the value in <tt>mappings.acme</tt>.
+     *
+     * @param code    the code to fetch the mapping for
+     * @param mapping the name of the mapping to fetch
+     * @return the value to use or an empty optional of either no mapping is present or the code is unknown
+     * @see #fetchMappingOrCode(String, String)
+     */
+    public Optional<String> fetchMapping(String code, String mapping) {
+        if (Strings.isEmpty(code)) {
+            return Optional.empty();
+        }
+
+        return performFetchField(normalizeCodeValue(code), mappingsField + "." + mapping);
+    }
+
+    /**
+     * Fetches the mapping for the give code or the code itself if no mapping is present or if the code is unknown.
+     *
+     * @param code    the code to fetch the mapping for
+     * @param mapping the name of the mapping to fetch
+     * @return either the mapping or the code itself if no mapping is present
+     * @see #fetchMapping(String, String)
+     */
+    @Nullable
+    public String fetchMappingOrCode(String code, String mapping) {
+        if (Strings.isEmpty(code)) {
+            return null;
+        }
+
+        return fetchMapping(code, mapping).orElseGet(() -> normalizeCodeValue(code));
+    }
+
+    /**
+     * Fetches either the primary or the secondary mapping stored for the given code.
+     * <p>
+     * This can be used to perform a two stage lookup. E.g. if an <tt>acme 1.0</tt> standard is requested, one could
+     * query <tt>acme-10</tt> as primary mapping and <tt>acme</tt> as secondary. This way, if a version dependent
+     * mapping is present, this will be used. Otherwise the base value from the standard is returned (if present).
+     *
+     * @param code             the code to fetch the mapping for
+     * @param primaryMapping   the more specific mapping to attempt to fetch
+     * @param secondaryMapping the more general mapping to fetch
+     * @return either the mapping for the first or second mapping to use or an empty optional if neither is present
+     */
+    public Optional<String> fetchMappings(String code, String primaryMapping, String secondaryMapping) {
```

Comment:
What is the approach if a value cease to exist after a specific mapping-version? Setting the key as obsolete will still resolve an older mapping? Or do we need to remove the "acme" and set the old value under all previous version of acme-## ?

Reply:
let's deal with this once we encounter this problem in the wild.... probably the suggested solution would be the way to go...